src/lib.rs

```diff
@@ -1013,7 +1013,17 @@ impl<'a> Cmd<'a> {
                 None => Stdio::null(),
             });
 
-            command.spawn().map_err(|err| Error::new_cmd_io(self, err))?
+            command.spawn().map_err(|err| {
+                // Try to determine whether the command failed because the current
+                // directory does not exist. Return an appropriate error in such a
+                // case.
+                if matches!(err.kind(), io::ErrorKind::NotFound) {
+                    if let Err(err) = self.shell.cwd.borrow().metadata() {
+                        return Error::new_current_dir(err);
```

Comment:
While we are at it, let’s include the directory itself into the r error message?

Reply:
Does what I just pushed look at all like what you were imagining?